lib/cli/file-set-pipeline/configure.js

```diff
@@ -28,16 +28,16 @@ var Configuration = require('../configuration');
  * @param {CLI} cli
  */
 function configure(cli) {
-    var configuration = new Configuration({
-        'detectRC': cli.detectRC,
-        'file': cli.configPath,
-        'settings': cli.settings,
-        'plugins': cli.plugins,
-        'output': cli.output,
-        'cwd': cli.cwd
-    });
-
-    cli.configuration = configuration;
+    if (!cli.configuration) {
```

Comment:
In fact, we can probably also wrap the traverser inside a similar if-statement (but keep the 'CLI.files' outside it of course), that should ensure mdastignore and mdastrc are treated similarly (as in, cached). Can you try that out?